packages/optimistic-oracle/src/proposer.js

```diff
@@ -73,60 +73,209 @@ class OptimisticOracleProposer {
+    }
+  }
 
-      // Pricefeed is either constructed correctly or is null.
-      if (!priceFeed) {
-        this.logger.error({
-          at: "OptimisticOracleProposer#sendProposals",
-          message: "Failed to construct a PriceFeed for price request",
-          priceRequest
-        });
-        continue;
-      }
+  // Submit disputes to proposed price requests.
+  async sendDisputes() {
+    this.logger.debug({
+      at: "OptimisticOracleProposer#sendDisputes",
+      message: "Checking for undisputed price requests to dispute"
+    });
 
-      // With pricefeed successfully constructed, get a proposal price
-      await priceFeed.update();
-      let proposalPrice;
-      try {
-        proposalPrice = (await priceFeed.getHistoricalPrice(priceRequest.timestamp)).toString();
-      } catch (error) {
-        this.logger.error({
-          at: "OptimisticOracleProposer#sendProposals",
-          message: "Failed to query historical price for price request",
-          priceRequest,
-          error
-        });
-        continue;
-      }
+    for (let priceRequest of this.optimisticOracleClient.getUndisputedProposals()) {
+      await this._sendDispute(priceRequest);
+    }
+  }
+
+  // Settle disputes where this bot was the disputer and proposals where this bot was the proposer.
+  async settleRequests() {
+    this.logger.debug({
+      at: "OptimisticOracleProposer#settleRequests",
+      message: "Checking for proposals and disputes to settle"
+    });
+
+    for (let priceRequest of this.optimisticOracleClient
+      .getSettleableProposals(this.account)
+      .concat(this.optimisticOracleClient.getSettleableDisputes(this.account))) {
+      await this._settleRequest(priceRequest);
+    }
+  }
+
+  /** **********************************
+   *
+   * INTERNAL METHODS
+   *
+   ************************************/
+
+  // Construct proposal transaction and send or return early if an error is encountered.
+  async _sendProposal(priceRequest) {
+    const priceFeed = await this._createOrGetCachedPriceFeed(priceRequest.identifier);
+
+    // Pricefeed is either constructed correctly or is null.
+    if (!priceFeed) {
+      this.logger.error({
+        at: "OptimisticOracleProposer#sendProposals",
+        message: "Failed to construct a PriceFeed for price request",
+        priceRequest
+      });
+      return;
+    }
+
+    // With pricefeed successfully constructed, get a proposal price
+    await priceFeed.update();
+    let proposalPrice;
+    try {
+      proposalPrice = (await priceFeed.getHistoricalPrice(priceRequest.timestamp)).toString();
+    } catch (error) {
+      this.logger.error({
+        at: "OptimisticOracleProposer#sendProposals",
+        message: "Failed to query historical price for price request",
+        priceRequest,
+        error
+      });
+      return;
+    }
+
+    // Create the transaction.
+    const proposal = this.ooContract.methods.proposePrice(
```

Comment:
Similar question about whether `ooContract` should expanded to `optimisticOracleContract`.

Reply:
Yup will do